addons/hr/models/hr_employee.py

```diff
@@ -535,6 +535,43 @@ def _get_tz(self):
                self.company_id.resource_calendar_id.tz or\
                'UTC'
 
+    def _get_calendar_tz(self, date_from=None):
+        # same as ``_get_tz`` but prioritze the date on the employee's calendar
+        self.ensure_one()
+        if date_from:  # contract overload has different signature
+            calendar = self._get_calendar(date_from)
+        else:
+            calendar = self._get_calendar()
+        return calendar.tz or\
+               self.tz
```

Comment:
oops you're right.
I think we should keep the method to create a notion of a "canoncial" timezone to get for an employee since there are so many....
I guess that was the point of `_get_tz` in the first place, but in some contexts it doesn't suit our needs and changing it now would break things.
What do you think?